### PR TITLE
Fix UPDATE_NEEDED is sent in case of unconsented device

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Add check that pt_exchanged_x_days_after_epoch is initialized,

https://github.com/smartdevicelink/sdl_core/issues/1215

### CODE REVIEW

**Comment by LitvinenkoIra** [comment]
@VVeremjova @okozlovlux @LevchenkoS please review

File: src/components/policy/policy_external/src/cache_manager.cc [hunk 1]
```diff
@@ -1032,22 +1032,28 @@ bool CacheManager::SetCountersPassedForSuccessfulUpdate(
   return true;
 }
 
-int CacheManager::DaysBeforeExchange(uint16_t current) {
+ReturnValue CacheManager::DaysBeforeExchange(uint16_t current) {
   LOG4CXX_AUTO_TRACE(logger_);
-  CACHE_MANAGER_CHECK(0);
+  CACHE_MANAGER_CHECK(kZero);
```
Comment (VVeremjova) [inline src/components/policy/policy_external/src/cache_manager.cc]: @LitvinenkoIra It is not necessary to change 0 to kZero here

**Comment by LitvinenkoIra** [comment]
Rebuild required

File: src/components/policy/policy_external/src/cache_manager.cc [hunk 2]
```diff
@@ -1035,16 +1035,22 @@ bool CacheManager::SetCountersPassedForSuccessfulUpdate(
 int CacheManager::DaysBeforeExchange(uint16_t current) {
   LOG4CXX_AUTO_TRACE(logger_);
   CACHE_MANAGER_CHECK(0);
+
+  const rpc::Optional<rpc::Integer<uint16_t, 0, 65535> > days_after_epoch =
```
Comment (LevchenkoS) [inline src/components/policy/policy_external/src/cache_manager.cc]: @LitvinenkoIra, perhaps better to use reference on `days_after_epoch` rather then copy

Comment (LitvinenkoIra) [inline src/components/policy/policy_external/src/cache_manager.cc]: @LevchenkoS Done
82e9111dce7b99cfc8e52124d9a40f47d229b980

File: src/components/policy/policy_external/src/cache_manager.cc [hunk 3]
```diff
@@ -1035,16 +1035,22 @@ bool CacheManager::SetCountersPassedForSuccessfulUpdate(
 int CacheManager::DaysBeforeExchange(uint16_t current) {
   LOG4CXX_AUTO_TRACE(logger_);
   CACHE_MANAGER_CHECK(0);
+
+  const rpc::Optional<rpc::Integer<uint16_t, 0, 65535> >& days_after_epoch =
+          (pt_->policy_table.module_meta->pt_exchanged_x_days_after_epoch);
+
+  if (!days_after_epoch->is_initialized()) {
+      return -1;
```
Comment (jacobkeeler) [inline src/components/policy/policy_external/src/cache_manager.cc]: Should be a two-space indent.

File: src/components/policy/policy_external/src/cache_manager.cc [hunk 4]
```diff
@@ -1035,16 +1035,22 @@ bool CacheManager::SetCountersPassedForSuccessfulUpdate(
 int CacheManager::DaysBeforeExchange(uint16_t current) {
   LOG4CXX_AUTO_TRACE(logger_);
   CACHE_MANAGER_CHECK(0);
+
+  const rpc::Optional<rpc::Integer<uint16_t, 0, 65535> >& days_after_epoch =
+          (pt_->policy_table.module_meta->pt_exchanged_x_days_after_epoch);
```
Comment (jacobkeeler) [inline src/components/policy/policy_external/src/cache_manager.cc]: Should be pushed back 4 spaces.

File: src/components/policy/policy_external/src/cache_manager.cc (hunk 3 above)
Comment (LitvinenkoIra) [inline src/components/policy/policy_external/src/cache_manager.cc]: @jacobkeeler Fixed ffa1498f4c2eb414232efa2d7b6d7297de650f90

File: src/components/policy/policy_external/src/cache_manager.cc (hunk 4 above)
Comment (LitvinenkoIra) [inline src/components/policy/policy_external/src/cache_manager.cc]: @jacobkeeler Fixed ffa1498f4c2eb414232efa2d7b6d7297de650f90

**Comment by LitvinenkoIra** [comment]
Rebuild required

**Pull request merged**